### PR TITLE
Completed file upload

### PR DESCRIPTION
**Pull request by Waqar-SE**
Image uploads work fine for items, not for orders yet.

### CODE REVIEW

File: frontend/components/invoices/ImageSelector.jsx [hunk 1]
```diff
@@ -7,23 +7,17 @@ import { useBillContext } from "@/context/BillContext";
...
 
-  const [success, setSuccess] = useState(false);
   const { dispatch } = useBillContext()
 
   useEffect(() => {
-    if (imageUrl && selectedFile) {
+    if (selectedFile) {
       const imageFile = new Image();
-      imageFile.src = imageUrl;
+      imageFile.src = URL.createObjectURL(selectedFile)
```
Comment (SalmanMughal789) [inline frontend/components/invoices/ImageSelector.jsx:19]: Can you please tell about this URL.createObjectURL ? Does it return base64 of a file ?

File: frontend/context/BillContext.jsx [hunk 2]
```diff
@@ -94,20 +95,21 @@ export function BillProvider({ children }) {
...
   const handleSaveInvoice = async () => {
+    // Add validation here, should only post if payload has correct data
     const { ok, data } = await useFetch('api/invoices/', {
       method: 'POST',
       body: JSON.stringify({
         company: state.selectedInvoice.company.id,
-        items: state.selectedInvoice.items.map(item => ({ ...item, object_id: item.objectId })),
-        orders: state.selectedInvoice.orders,
+        items: state.selectedInvoice.items.map(item => ({ ...item, object_id: item.objectId })).filter(item => item.amount !== 0),
+        orders: state.selectedInvoice.orders.filter(order => order.items.length > 0),
```
Comment (SalmanMughal789) [inline frontend/context/BillContext.jsx:104]: Also there should be no item inside an order which have an amount of zero.

**Review by SalmanMughal789 — COMMENTED** [review]
Yes it looks good. Please merge them into master branch.

File: frontend/components/invoices/ImageSelector.jsx (hunk 1 above)
Comment (Waqar-SE) [reply frontend/components/invoices/ImageSelector.jsx:19]: It's saving us the decryption of image to base64 when showing image from the backend. 
The file like is provided from the backend and it's downloaded as user hovers over it. 

From the frontend we post iimage, it stores it user's local system as a object with `URL.createObjectURL`. 
In this way the image popover is always showing image from either the remote server or the local server.

File: frontend/context/BillContext.jsx (hunk 2 above)
Comment (Waqar-SE) [reply frontend/context/BillContext.jsx:125]: Yes, we need to work on complete validation.

**Pull request merged**